### ISSUE TITLE
Can this be used in Regression Problem?

### ISSUE THREAD

**Issue by err-or** [CuthbertCai/pytorch_DANN#16]
Hi @tengerye @CuthbertCai! Thank you for sharing the code. 

I was wondering if this particular domain adaptation technique can be used in a regression problem. Let's say if I want to use this domain adaptation technique for a regression problem where the loss function for the main task is MAE or MSE. I am a bit confused about how to actually backpropagate the loss in such a case. Since in the classification problem, the total loss is a summation (according to the code) of the class loss and domain loss and as both of them are negative log-likelihood loss there is no issue while adding them. However, if the loss for the actual task and the loss for the domain classification task are different what should we do in that case? Your insight would be really helpful! Thanks!

**Comment by CuthbertCai** [comment]
Thanks for your appreciation! I'm not familiar with the regression problem. But I think this adversarial technique can be directly used in a regression problem if you also adopt a neural network in your task. This technique is conducted in the feature space, so I think maybe you can just add the adversarial loss to an intermediate feature space. As for combining the task loss and the adversarial loss, I think we can directly add them together. Besides, maybe a hyper-parameter is needed to balance different losses.

**Comment by CuthbertCai** [comment]
> Hi @tengerye @CuthbertCai! Thank you for sharing the code.
> 
> I was wondering if this particular domain adaptation technique can be used in a regression problem. Let's say if I want to use this domain adaptation technique for a regression problem where the loss function for the main task is MAE or MSE. I am a bit confused about how to actually backpropagate the loss in such a case. Since in the classification problem, the total loss is a summation (according to the code) of the class loss and domain loss and as both of them are negative log-likelihood loss there is no issue while adding them. However, if the loss for the actual task and the loss for the domain classification task are different what should we do in that case? Your insight would be really helpful! Thanks!

Thanks for your appreciation! I'm not familiar with the regression problem. But I think this adversarial technique can be directly used in a regression problem if you also adopt a neural network in your task. This technique is conducted in the feature space, so I think maybe you can just add the adversarial loss to an intermediate feature space. As for combining the task loss and the adversarial loss, I think we can directly add them together. Besides, maybe a hyper-parameter is needed to balance different losses.
